Code/Legacy/CrySystem/DebugCallStack.cpp

```diff
@@ -615,45 +596,55 @@ void DebugCallStack::LogExceptionInfo(EXCEPTION_POINTERS* pex)
-        // if you're customizing the engine this is also your opportunity to deal with it.
-        if (g_cvars.sys_no_crash_dialog != 0 || !g_bUserDialog)
+        AZStd::string msg = AZStd::string::format(
+            "O3DE has encountered an unexpected error.\n\nDo you want to manually report the issue on Github ?\nInformation about the "
+            "crash are located in %s via error.log and error.dmp",
+            path.c_str());
+        constexpr bool showCancel = false;
+        AZStd::string res = nativeUI->DisplayYesNoDialog("O3DE unexpected error", msg, showCancel);
+        if (res == "Yes")
         {
-            // ------------ place custom crash handler here ---------------------
-            // it should launch an executable!
-            /// by  this time, error.bmp will be in the engine root folder
-            // error.log and error.dmp will also be present in the engine root folder
-            // if your error dumper wants those, it should zip them up and send them or offer to do so.
-            // ------------------------------------------------------------------
+            AZStd::wstring arg(path.begin(), path.end());
+            ShellExecuteW(nullptr, L"open", arg.c_str(), NULL, NULL, SW_SHOWNORMAL);
+
+            AZStd::string reportIssueUrl;
+            if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
+                settingsRegistry->Get(reportIssueUrl, SettingKey_IssueReportLink);
+
+            if (reportIssueUrl.empty())
+                reportIssueUrl = IssueReportLinkFallback;
+
+            arg = AZStd::wstring(reportIssueUrl.begin(), reportIssueUrl.end());
+            ShellExecuteW(nullptr, L"open", arg.c_str(), NULL, NULL, SW_SHOWNORMAL);
         }
     }
+
     const bool bQuitting = !gEnv || !gEnv->pSystem || gEnv->pSystem->IsQuitting();
 
-    //[AlexMcC|16.04.10] When the engine is shutting down, MessageBox doesn't display a box
-    // and immediately returns IDYES. Avoid this by just not trying to save if we're quitting.
-    // Don't ask to save if this isn't a real crash (a real crash has exception pointers)
     if (g_cvars.sys_no_crash_dialog == 0 && g_bUserDialog && gEnv->IsEditor() && !bQuitting && pex)
     {
         BackupCurrentLevel();
 
-        const INT_PTR res = DialogBoxParam(gDLLHandle, MAKEINTRESOURCE(IDD_CONFIRM_SAVE_LEVEL), NULL, DebugCallStack::ConfirmSaveDialogProc, NULL);
-        if (res == IDB_CONFIRM_SAVE)
+        if (auto nativeUI = AZ::Interface<AZ::NativeUI::NativeUIRequests>::Get(); nativeUI != nullptr)
         {
-            if (SaveCurrentLevel())
-            {
-                MessageBoxW(NULL, L"Level has been successfully saved!\r\nPress Ok to terminate Editor.", L"Save", MB_OK);
-            }
-            else
+            constexpr bool showCancel = false;
+            AZStd::string res = nativeUI->DisplayYesNoDialog(
+                "Save your changes ?",
```

Comment:
nit: Remove the space before '?'
```suggestion
                "Save your changes?",
```